src/DomainSearch.js

```diff
@@ -173,6 +171,12 @@ export default class DomainSearch extends Component {
     } = this.state;
 
     const domainCount = selectedDomains.length;
+    const hasExactMatch = results && results.exactMatchDomain && results.exactMatchDomain.available;
+
+    window.onbeforeunload = () => {
+      // Most browsers control the return message to the user, we can safely return an empty string here.
+      return hasExactMatch || domainCount > 0 ? '' : undefined;
```

Comment:
I think we should allow the reseller to enter a custom message.  We can default to it to empty string though.

Reply:
From https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload:

"Starting with Firefox 44, Chrome 51, Opera 38 and Safari 9.1, a generic string not under the control of the webpage will be shown instead of the returned string."

That would leave a _very_ small subset of modern browsers that would even respect a custom user-provided message.